R/expect_lint.R

```diff
@@ -59,11 +62,11 @@ expect_lint <- function(content, checks, ..., file = NULL, language = "en") {
 
   lints <- lint(file, ...)
   n_lints <- length(lints)
-  lint_str <- if (n_lints) paste(c("", lints), collapse = "\n") else ""
+  lint_fmt <- function() if (n_lints > 0L) paste(c("", lints), collapse = "\n") else ""
```

Comment:
Why do we need this?

Reply:
pretty edge-casey, but if there are 100s of lints and this is `expect_no_lint()`, we don't need to generate this big string, we only report "N lints when 0 were expected".

Reply:
I don't see a code path where it's skipped?
The `is.null(checks)` path includes the list of lints at the end of the message, no?

Reply:
Indeed... 🤦